astropy/coordinates/polarization.py

```diff
@@ -0,0 +1,235 @@
+    -6: StokesSymbol("YY", "Y parallel linear: <YY*>"),
+    -7: StokesSymbol("XY", "XY cross linear: Re(<XY*>)"),
+    -8: StokesSymbol("YX", "YX cross linear: Im(<XY*>)"),
+}
+
+STOKES_VALUE_SYMBOL_MAP = FITS_STOKES_VALUE_SYMBOL_MAP.copy()
+UNKNOWN_STOKES_VALUE = -99999
+
+
+@contextmanager
+def custom_stokes_symbol_mapping(
+    mapping: Dict[int, StokesSymbol], replace: bool = False
+):
+    """
+    Add a custom set of mappings from values to Stokes symbols.
+
+    Parameters
+    ----------
+    mappings
+        A list of dictionaries with custom mappings between values (integers)
+        and `.StokesSymbol` classes.
+    replace
+        Replace all mappings with this one.
+    """
+    global STOKES_VALUE_SYMBOL_MAP
+
+    original_mapping = STOKES_VALUE_SYMBOL_MAP.copy()
+    if not replace:
+        STOKES_VALUE_SYMBOL_MAP = {**original_mapping, **mapping}
+    else:
+        STOKES_VALUE_SYMBOL_MAP = mapping
+
+    yield
+
+    STOKES_VALUE_SYMBOL_MAP = original_mapping
+
+
+class StokesCoordInfo(MixinInfo):
+    # The attributes containing actual information.
+    _represent_as_dict_attrs = {"value"}
+    # Since there is only one attribute, use a column with the name to represent it
+    # (rather than as name.value)
+    _represent_as_dict_primary_data = "value"
+    # Attributes that should be presented as positional arguments to
+    # the class initializer (which takes "stokes" as an argument, not "value").
+    _construct_from_dict_args = ("value",)
+
+    @property
+    def unit(self):
+        return None
+
+    @property
+    def dtype(self):
+        return self._parent._data.dtype
+
+    @staticmethod
+    def default_format(val):
+        return f"{val.symbol}"
+
+    def new_like(self, cols, length, metadata_conflicts="warn", name=None):
+        """
+        Return a new StokesCoord instance which is consistent with the
+        input ``cols`` and has ``length`` rows.
+
+        This is intended for creating an empty column object whose elements can
+        be set in-place for table operations like join or vstack.
+
+        Parameters
+        ----------
+        cols : list
+            List of input columns
+        length : int
+            Length of the output column object
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts
+        name : str
+            Output column name
+
+        Returns
+        -------
+        col : `~astropy.coordinates.StokesCoord` (or subclass)
+            Empty instance of this class consistent with ``cols``
+
+        """
+        # Get merged info attributes like shape, dtype, format, description, etc.
+        attrs = self.merge_cols_attributes(
+            cols, metadata_conflicts, name, ("meta", "format", "description")
+        )
+
+        # Make an empty StokesCoord.
+        shape = (length,) + attrs.pop("shape")
+        data = np.empty(shape=shape, dtype=self.dtype)
```

Comment:
Here, I fear it is me copying stuff in without tests... I have no time immediately to test but think (from QuantityInfo), that what you would do here is write `dtype=attrs.pop("dtype")`.

Reply:
Ah that's moved the problem, I am now getting `TypeError: 'StokesCoord' object does not support item assignment`. How do I make `ShapedLikeNDArray` support item assignment?

Reply:
Also other than vstack is there any other table operations I should test?

Reply:
Ah, yes, the next whack-a-mole... Since item assignment can be trickier, you need to define it yourself. Fortunately, not difficult for this case:
```
def __setitem__(self, item, value):
    self._data[item] = self.__class__(value)._data
```